### PR TITLE
Mix command to check for (and cleanup) unused snapshots

### PR DESCRIPTION
**Pull request by SergioInToronto**
## Purpose

* `mix ink_snap.check`
* `mix ink_snap.check --delete`

## Testing

`mix ink_snap.check`

```
Found 4 unused snapshot(s):
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_collection_create_2_will_create_a_collection.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_collection_update_2_will_update_a_collection.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_subspace_create_2_will_create_a_subspace.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_subspace_update_2_will_update_a_subspace.snap

Remove them by running:

    mix ink_snap.check --delete
```

`mix ink_snap.check --delete`

```
Removed 4 unused snapshot(s):
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_collection_create_2_will_create_a_collection.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_collection_update_2_will_update_a_collection.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_subspace_create_2_will_create_a_subspace.snap
  test/_snapshots/inkio_rpc/servers/namespace_server_test/test_subspace_update_2_will_update_a_subspace.snap
```

* Return code `1` means unused snapshots were found
* Takes ~10 seconds on InkIO, ~11s in Helios

## AI Plan

<details>
<summary>Click to expand</summary>

# Plan: Detect unused snapshot files (`mix ink_snap.check`)

## Goal

Add a shippable mix task that detects stale/orphaned `.snap` files (left behind
when a test is renamed and `SNAPSHOT_UPDATE=true` regenerates snapshots under
the new name). Runnable in CI to fail on stale snapshots. Optional cleanup.

## Why this approach (static compile-time introspection, no test execution)

Evidence gathered:

- `test_grpc` (ink-io) and `test_snapshot` (ink-snap) are **wrapper macros**
  that both expand to a plain ExUnit `test` calling `InkSnap.handle_snapshot!`
  (see `deps/movableink/lib/movableink_grpc/snap_helpers.ex:78-85`). A naive
  "grep for `test_snapshot` calls" AST walk finds **zero** snapshot tests in
  ink-io. Wrapper macros are unknowable without expansion.
- The snapshot filename derives **only** from the composed ExUnit name
  (`"test #{describe} #{name}"`, `ex_unit/case.ex:701/705`) fed through
  `InkSnap`'s `snapshot_file_basename/1`. The wrapper macro name is irrelevant.
- Compiling a test file (`Code.require_file`) **expands all macros** and makes
  ExUnit record every final test as `%ExUnit.Test{}` in the module, readable via
  `module.__ex_unit__().tests` (`ex_unit/case.ex:629-637, 727-728`). This is a
  **dry run**: modules compile, tests do NOT execute.
- Each `%ExUnit.Test{}` carries `.name` (the atom = runtime `env.function`,
  already truncated by `build_test_name` if >255 bytes, `case.ex:922`) and
  `.tags.file` (source file). Both are exactly what `InkSnap.snapshot_file!/2`
  needs. Using `.name` (not `.description`) matches runtime filenames byte-for-byte.

Result: expected `.snap` set is computed correct-by-construction for ANY wrapper
macro, reusing InkSnap's own path logic. No suite run, no name-composition
reimplementation, no AST fragility.

### 1. `lib/ink_snap.ex` — expose minimal internals (small refactor)

The task needs InkSnap's path + naming logic. Rather than duplicate, extract a
thin public (`@doc false`) helper and reuse existing private fns.

- Add `@doc false def snapshot_file_for(file, function_name)` that wraps the
  existing private `snapshot_file!/2`. (`snapshot_file!` already does everything:
  resolves test paths via cached config, builds
  `<test_path>/<snap_dir>/<sub_path>/<basename>`.) One-line delegation.
- Add `@doc false def snapshot_dirs()` returning, per resolved test path, the
  snapshot root dir `Path.join(test_path, snapshot_directory)` — the set of
  directories to scan for existing `.snap` files. Reuses `test_paths/0` +
  `snapshot_directory/0` (already private; expose via this wrapper only).
- No behavior change to existing runtime path. `snapshot_file!/2` already calls
  `Mix.Project.get!()` / `Mix.Project.config()`, which are available in a mix
  task context.

`ponytail:` reuse existing private fns, add only thin `@doc false` wrappers — no
new path logic.

### 2. `lib/mix/tasks/ink_snap.check.ex` — new mix task (ships in package)

`Mix.Task` named `ink_snap.check`. Algorithm:

1. `Mix.Task.run("compile")` then `Application.ensure_all_started(:ink_snap)`.
2. `ExUnit.start(autorun: false)` — required so `__after_compile__` registration
   (`case.ex:650-661`) doesn't raise; `autorun: false` prevents any run.
3. Load `test/test_helper.exs` if present (case templates / support live there),
   guarded by `File.exists?`. Rationale: test files `use SomeCase` from support.
4. Discover test files: for each dir from `InkSnap.snapshot_dirs()`'s parent test
   paths, glob `**/*_test.exs`. (Use `Mix.Project.config()[:test_paths]` /
   default `test`, mirroring InkSnap's own resolution.)
5. `Code.require_file/1` each test file (compile-only dry run). Collect all
   loaded ExUnit modules — track modules before/after, or scan required files'
   defined modules — and read `mod.__ex_unit__().tests`.
6. For each `%ExUnit.Test{name: name, tags: %{file: file}}`, compute expected
   path via `InkSnap.snapshot_file_for(file, name)`. Build `expected` MapSet.
7. Discover `existing` on disk: for each `InkSnap.snapshot_dirs()`, glob
   `**/*.snap` (only `*.snap`, nothing else ever touched).
8. `orphans = existing -- expected`.
9. Report:
   - No orphans → print OK, exit 0.
   - Orphans, no `--clean` → print each orphan path, print count, exit 1.
   - Orphans, `--clean` → delete each orphan (`File.rm!`), then prune now-empty
     snapshot subdirs (bottom-up `File.rmdir` ignoring non-empty), print what
     was removed, exit 0.

Flags: `--clean` (delete + prune empty dirs).

Safety: only files matching `*.snap` under resolved snapshot dirs are ever
considered or deleted. Non-`.snap` files ignored entirely.

Edge cases handled:
- Test excluded only by runtime tag filters still compiles → still counted (no
  false positive; this is the win over runtime tracking).
- Truncated long test names: `.name` atom already truncated identically to
  runtime → exact match.
- Multiple test paths / custom `:snapshot_directory` config → reuses InkSnap
  resolution, so consistent.
- `test/test_helper.exs` missing → skipped gracefully.

### 3. `mix.exs` — wire into CI

- Add `"ink_snap.check"` to the `ci` alias (`mix.exs:39-46`) so this repo's own
  CI gates on stale snapshots.
- `files:` list already includes `lib` (`mix.exs:83`), so
  `lib/mix/tasks/ink_snap.check.ex` ships in the hex package automatically —
  ink-io and other consumers get `mix ink_snap.check` for free. No change needed
  to `files:`.

### 4. `README.md` — document

Add a short "Detecting unused snapshots" section: `mix ink_snap.check`,
`mix ink_snap.check --clean`, and a CI usage note.

## Testing

- `test/mix/tasks/ink_snap.check_test.exs`: create a temp project fixture (or
  use this repo's own `test/_snapshots/`) with a known-orphan `.snap` file;
  assert the task detects it and exits non-zero; assert `--clean` removes it and
  prunes the empty dir; assert clean repo reports OK / exit 0.
- Manual: drop a bogus `test/_snapshots/ink_snap_test/test_bogus.snap`, run
  `mix ink_snap.check` → expect it flagged; `--clean` → expect it gone.

## Risks / open notes

- Compiling test files runs any **compile-time** (not test-time) side effects in
  those files. Standard ExUnit test files have none; acceptable and unavoidable
  for a dry run. Documented.
- Requires the project's test support (case templates) to be loadable — handled
  by compiling first + loading `test_helper.exs`.
- Collecting "which modules were defined by a required file": use
  `Code.require_file/1` return value (list of `{module, binary}`) to get exactly
  the modules per file — robust, no global module diffing needed.

</details>

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-erlang 27.2
+erlang 27.3.4
```
Comment (SergioInToronto) [inline .tool-versions:1]: bump to fix TLS known issue in older versions. [See failing CI](https://github.com/coherentpath/ink-snap/actions/runs/28612006768/job/84846516224?pr=26)
